### PR TITLE
Compatibility validator improvements

### PR DESCRIPTION
**Pull request by qwwdfsad**
* ExternalApi annotation to mark declarations that stdlib depends on
* New nonPublicMarkers API for filtering out effectively internal API

### CODE REVIEW

File: src/main/kotlin/api/KotlinSignaturesLoading.kt [hunk 1]
```diff
@@ -47,14 +46,34 @@ fun Sequence<InputStream>.loadApiFromJvmClasses(visibilityFilter: (String) -> Bo
...
-                    metadata.isFileOrMultipartFacade() || isDefaultImpls(metadata)
+                    metadata.isFileOrMultipartFacade() || isDefaultImpls(metadata),
+                    annotations(visibleAnnotations, invisibleAnnotations)
                 )
         }}
         .asIterable()
         .sortedBy { it.name }
 }
 
+fun List<ClassBinarySignature>.filterOutAnnotated(targetAnnotations: Set<String>): List<ClassBinarySignature> {
```
[the remark this answers is not in the document]
Comment (qwwdfsad) [reply src/main/kotlin/api/KotlinSignaturesLoading.kt:57]: Indeed, but I'd prefer to keep it consistent and aligned with `filterOutNonPublic`. Doesn't really matter for a small internal method

File: src/main/kotlin/api/KotlinSignaturesLoading.kt [hunk 2]
```diff
@@ -47,14 +46,34 @@ fun Sequence<InputStream>.loadApiFromJvmClasses(visibilityFilter: (String) -> Bo
...
+    if (targetAnnotations.isEmpty()) return this
+    return filter {
```
[the remark this answers is not in the document]
Comment (qwwdfsad) [reply src/main/kotlin/api/KotlinSignaturesLoading.kt:59]: This cost should be pretty marginal for BCV. 
As a bonus, current signatures reading code (https://github.com/Kotlin/binary-compatibility-validator/blob/bcv-imporvements/src/main/kotlin/KotlinApiBuildTask.kt#L36) is simple and straightforward

[the remark this answers is not in the document]
Comment (ilya-g) [reply src/main/kotlin/api/KotlinSignaturesLoading.kt:58]: Ok. FTR non public api is filtered out both in loadApiFromJvmClasses and in filterOutNonPublic

**Review by ilya-g — APPROVED** [review]

**Comment by SergejIsbrecht** [comment]
Are there any plans to commit this and release a new version? This is acutally what I was looking for to exclude certain classes, which are public but shound not be "public" (e.g. BuildConfig)

**Pull request merged**